src/app/server/Server.cpp

```diff
@@ -126,6 +126,7 @@ CHIP_ERROR Server::Init(const ServerInitParams & initParams)
     VerifyOrExit(initParams.operationalKeystore != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrExit(initParams.opCertStore != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrExit(initParams.reportScheduler != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(initParams.dataModelProvider != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
Since this will likely break every server application out there, we should have clearer logging here about what exactly failed so people can fix it.  Right now in a typical small-device config where file/line are not attached to CHIP_ERROR there will be no indication what exactly failed...

Reply:
Added logs as well as a note in our upgrading document.